pkg/preparation/sources/sources.go

```diff
@@ -23,13 +22,19 @@ func (e ErrUnrecognizedSourceKind) Error() string {
 // API is the API for accessing and managing sources.
 type API struct {
 	Repo Repo
+	// GetLocalFSForPathFn is called to get an fs.FS for a [model.LocalSourceKind] source.
```

Comment:
curious why not just use os.DirFS?

nb, but best I can tell this is used for testing. I would argue that afero IS a different source kind, and actually should be defined as such. That would mean instead of having a seperate depedency, you'd just have another case statement. could be called model.TestSourceKind or something to that effect.

Reply:
maybe that is harder to implement than it looks